packages/verified-fetch/test/utils/parse-url-string.spec.ts

```diff
@@ -357,15 +566,275 @@ describe('parseUrlString', () => {
+          protocol: 'ipns',
+          cid: cid.toString(),
+          path: 'path/to/dir/',
+          query: {
+            format: 'tar'
+          }
+        }
+      )
+    })
+  })
+
+  HTTP_PROTOCOLS.forEach(proto => {
+    describe(`${proto}://example.com/ipfs/<CID> URLs`, () => {
+      let peerId: PeerId
+      let cid: CID
+
+      beforeEach(async () => {
+        peerId = await createEd25519PeerId()
+        cid = CID.parse('QmdmQXB2mzChmMeKY47C43LxUdg1NDJ5MWcKMKxDu7RgQm')
+        ipns.resolve.withArgs(matchPeerId(peerId)).resolves({
+          cid,
+          path: ''
+        })
+      })
+
+      it('should parse an IPFS Gateway URL with a CID only', async () => {
+        await assertMatchUrl(
+          `${proto}://example.com/ipns/${peerId}`, {
+            protocol: 'ipns',
+            cid: cid.toString(),
+            path: '',
+            query: {}
+          }
+        )
+      })
+
+      it('can parse an IPNS Gateway URL with CID+path', async () => {
+        await assertMatchUrl(
+          `${proto}://example.com/ipns/${peerId}/1 - Barrel - Part 1/1 - Barrel - Part 1 - alt.txt`, {
+            protocol: 'ipns',
+            cid: cid.toString(),
+            path: '1 - Barrel - Part 1/1 - Barrel - Part 1 - alt.txt',
+            query: {}
+          }
+        )
+      })
+
+      it('can parse an IPNS Gateway URL with CID+directoryPath', async () => {
+        await assertMatchUrl(
+          `${proto}://example.com/ipns/${peerId}/path/to/dir/`, {
+            protocol: 'ipns',
+            cid: cid.toString(),
+            path: 'path/to/dir/',
+            query: {}
+          }
+        )
+      })
+
+      it('can parse an IPNS Gateway URL with CID+queryString', async () => {
+        await assertMatchUrl(
+          `${proto}://example.com/ipns/${peerId}?format=car`, {
+            protocol: 'ipns',
+            cid: cid.toString(),
+            path: '',
+            query: {
+              format: 'car'
+            }
+          }
+        )
+      })
+
+      it('can parse an IPNS Gateway URL with CID+path+queryString', async () => {
+        await assertMatchUrl(
+          `${proto}://example.com/ipns/${peerId}/1 - Barrel - Part 1/1 - Barrel - Part 1 - alt.txt?format=tar`, {
+            protocol: 'ipns',
+            cid: cid.toString(),
+            path: '1 - Barrel - Part 1/1 - Barrel - Part 1 - alt.txt',
+            query: {
+              format: 'tar'
+            }
+          }
+        )
+      })
+
+      it('can parse an IPNS Gateway URL with CID+directoryPath+queryString', async () => {
+        await assertMatchUrl(
+          `${proto}://example.com/ipns/${peerId}/path/to/dir/?format=tar`, {
+            protocol: 'ipns',
+            cid: cid.toString(),
+            path: 'path/to/dir/',
+            query: {
+              format: 'tar'
+            }
+          }
+        )
+      })
+    })
+  })
+
+  HTTP_PROTOCOLS.forEach(proto => {
```

Comment:
I was `CTRL+F`ing for `http` and `https` and it took me way too long to find this.